### PR TITLE
Add android enchantments

### PR DESCRIPTION
**Pull request by nikolapeevski-mentormate**
Adds the following android enchantments:

- Loading indicator on initial load.
- Loading indicator on when buffering.
- Loading indicator and a timer if the connection has been lost.
- On screen back button.

### CODE REVIEW

File: src/android/BrightcoveActivity.java [hunk 1]
```diff
@@ -26,17 +30,74 @@
...
         setContentView(this.getIdFromResources(BRIGHTCOVE_ACTIVITY_NAME, LAYOUT_VIEW_KEY));
         brightcoveVideoView = (BrightcoveExoPlayerVideoView) findViewById(this.getIdFromResources(BRIGHTCOVE_VIEW_NAME, ID_VIEW_KEY));
+        this.progressBar = findViewById(this.getIdFromResources("progressBar", ID_VIEW_KEY));
+        this.button = findViewById(this.getIdFromResources("button1", ID_VIEW_KEY));
+        this.button.setOnClickListener(e -> {
+            if (this.offline) {
+                this.timeout.cancel();
+            }
+            this.finish();
+            this.sendCallback(BACK_STATUS);
```
Comment (joesbigidea) [inline src/android/BrightcoveActivity.java:52]: I noticed there's an onBackPressed method in this class. Is that triggered by the Android back button or something? If so should it be cancelling the timeout as well? 

I see the "send callback, finish()" idiom all over in here. I wonder if that should be a method that always cancels the timer if it is not null.

File: src/android/BrightcoveActivity.java [hunk 2]
```diff
@@ -26,17 +30,74 @@
 
     private static final String BACK_STATUS = "closed";
     private static final String COMPLETE_STATUS = "completed";
+    private static final String OFFLINE_STATUS = "OFFLINE";
+
+    private ProgressBar progressBar;
+    private Button button;
```
Comment (joesbigidea) [inline src/android/BrightcoveActivity.java:36]: Can you put a more descriptive name on this button?

File: src/android/BrightcoveActivity.java (hunk 1 above)
Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:52]: Yep, I'll concat both calls in a single method and address the cancelation in other places.

File: src/android/BrightcoveActivity.java [hunk 3]
```diff
@@ -26,17 +30,74 @@
...
+
+        eventEmitter.on("bufferingStarted", event -> {
+            this.progressBar.setVisibility(View.VISIBLE);
+        });
+
+        eventEmitter.on("error", event -> {
+            if (!this.offline) {
+                this.offline = true;
+                this.progressBar.setVisibility(View.VISIBLE);
+                this.timeout = new CountDownTimer(60000, 5000) {
```
Comment (joesbigidea) [inline src/android/BrightcoveActivity.java:82]: what happens after 60 seconds?

Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:82]: The activity closes if playing has not resumed.

File: src/android/BrightcoveActivity.java (hunk 2 above)
Comment (nikolapeevski-mentormate) [reply src/android/BrightcoveActivity.java:36]: Yep.

**Review by joesbigidea — APPROVED** [review]

**Pull request merged**